spec/support/shared_examples/step_controller.rb

```diff
@@ -9,6 +9,18 @@
 
       expect(step).to be_an_instance_of step_class
     end
+
+    context "no application is currently under way" do
+      it "redirects to the homepage" do
+        if subject.class != IntroduceYourselfController
+          session[:snap_application_id] = nil
+
+          get :edit
+
+          expect(response).to redirect_to(root_path)
+        end
```

Comment:
On the fence about wrapping this conditional around here to exclude just that one controller.